### ISSUE TITLE
Which version of the noise_index has been employed in your model?

### ISSUE THREAD

**Issue by ppj567** [xu5zhao/BiCro#1]
Hi, thank you for your great work! 

I am a little bit confused that the noise_index adopted in NCR and DECL are different, while in BiCro it seems to take the noise_index in NCR in the file [./noise_index](https://github.com/xu5zhao/BiCro/tree/main/noise_index). However, in the training script, I cannot find the referred directory 'noise_index/coco_precomp_0.4.npy'. https://github.com/xu5zhao/BiCro/blob/0aee28bd600f652ac3d70d5efc5eecdde65f36f5/train_coco.sh#L23
In fact, the noise_index is quite important in comparing with different methods. For instance, if DECL adopts the same noise_index as NCR, it will only show a tiny edge over NCR. This phenomenon could be found at [their repo](https://github.com/QinYang79/DECL).

Therefore, in order to make a thorough comparison, could you share the noise_index adopted in BiCro? Thanks a lot.

**Comment by xu5zhao** [comment]
hi, We did use the same noise Settings as NCR, and we uploaded two files ('noise_index/f30k_precomp_0.4' and 'noise_index/coco_precomp_0.4.npy')

**Comment by ppj567** [comment]
Hi, NCR only provides the noise_index of 20% and 50% ratios in their repo. Did you mean that you adopted the noise_index of 20% ratio from NCR, and adopted that of 40% and 60% ratios from DECL?

**Comment by ppj567** [comment]
> 

By the way, I trained BiCro on Flickr30K with the noise ratio of 50% (the noise_index is taken from DECL), and I obtained the following results (the best training epoch is 36), which are inferior to NCR and DECL. Since I did not change any part of BiCro, could you give some advise on how to reproduce a good result? Thanks a lot.

![image](https://user-images.githubusercontent.com/78494473/232402686-935173f6-4491-4f95-97ca-03c228e9ce6b.png)
![image](https://user-images.githubusercontent.com/78494473/232402849-dd5818a8-9c18-4e58-ac25-801fda7ea3a7.png)

**Comment by ppj567** [comment]
> hi, We did use the same noise Settings as NCR, and we uploaded two files ('noise_index/f30k_precomp_0.4' and 'noise_index/coco_precomp_0.4.npy')

Hi, it seems the rest two files are still missing ('noise_index/f30k_precomp_0.6.npy' and 'noise_index/coco_precomp_0.6.npy').

**Comment by xu5zhao** [comment]
> Hi, NCR only provides the noise_index of 20% and 50% ratios in their repo. Did you mean that you adopted the noise_index of 20% ratio from NCR, and adopted that of 40% and 60% ratios from DECL?

We directly adopt the noise index of 20% ratios from NCR, and generate the noise index of 40% and 60% ratios by NCR method.

**Comment by xu5zhao** [comment]
> we uploaded two files ('noise_index/f30k_precomp_0.4' and 'noise_index/coco_precomp_0.4.npy')

**Comment by xu5zhao** [comment]
> > hi, We did use the same noise Settings as NCR, and we uploaded two files ('noise_index/f30k_precomp_0.4' and 'noise_index/coco_precomp_0.4.npy')
> 
> Hi, it seems the rest two files are still missing ('noise_index/f30k_precomp_0.6.npy' and 'noise_index/coco_precomp_0.6.npy').

 we uploaded two files ('noise_index/f30k_precomp_0.6' and 'noise_index/coco_precomp_0.6.npy')

**Comment by xu5zhao** [comment]
> > 
> 
> By the way, I trained BiCro on Flickr30K with the noise ratio of 50% (the noise_index is taken from DECL), and I obtained the following results (the best training epoch is 36), which are inferior to NCR and DECL. Since I did not change any part of BiCro, could you give some advise on how to reproduce a good result? Thanks a lot.
> 
> ![image](https://user-images.githubusercontent.com/78494473/232402686-935173f6-4491-4f95-97ca-03c228e9ce6b.png) ![image](https://user-images.githubusercontent.com/78494473/232402849-dd5818a8-9c18-4e58-ac25-801fda7ea3a7.png)

We haven't tried BiCro on Flickr30K with the noise ratio of 50% in the DECL setting way.  Perhaps you can try setting the parameter '--fit_type gmm', It behaves steadily under "--p_threshold", default=0.5, type=float, help="clean probability threshold" based on our previous experience.

**Comment by ppj567** [comment]
Hi, I tried to reproduce the results of BiCro on Flickr30k with 40% noise rate based on your newly updated noise_index files. During training, I kept all the settings unchanged with yours, and obtained the following results. BiCro performs a little bit better than DECL, but there is still a gap between the reported numbers in the original paper. Should I do some further  modifications so as to boost the performance, or run a couple of more times to find the best one? Thanks.

![image](https://user-images.githubusercontent.com/78494473/232959136-898d45f8-514b-4c76-aa37-3b784637ff08.png)
![image](https://user-images.githubusercontent.com/78494473/232959815-87e7eacd-c008-48e6-b13d-558f1edc606e.png)

**Comment by ppj567** [comment]
https://github.com/xu5zhao/BiCro/blob/bd36643537952b6ee1a20f59e079ccf3ffbaff0b/co_train.py#L339

I found something inconsistent with the original paper, that during the first 10 epochs of co-learning, the clean data is optimized using soft labels c_y rather than hard labels (i.e, 1). In order to reproduce the numbers reported in the paper, should I change soft labels into hard labels just as the paper stated?

**Comment by ppj567** [comment]
I ran multiple times with different seeds on Flickr30k with 40% noise rate and obtained very close results as follows:

![image](https://user-images.githubusercontent.com/78494473/233528835-05625d0b-24b7-4e58-9f5b-36600af678d5.png)


I also tried to use hard labels for clean data just as the paper stated, and found that it will lead to much worse results (up to -6% decreases). Could the authors share the checkpoints of the 10-epoch warm-up models and the best co-learning models? Thanks a lot.

**Comment by xu5zhao** [comment]
The results in this paper are based on SGRAF, which includes two models: SGR and SAF. You seem to have only tested the SGR. SGRAF is more accurate, but requires training both models. I uploaded the checkpoints training in on Flickr30k with 40% noise rate.

**Comment by xu5zhao** [comment]
I updated the evaluation.py to fix the bug, noting that the path parameters for both models: --mode_path and --mode_path1 are required when testing the SGRAF model.

**Comment by ppj567** [comment]
> The results in this paper are based on SGRAF, which includes two models: SGR and SAF. You seem to have only tested the SGR. SGRAF is more accurate, but requires training both models. I uploaded the checkpoints training in on Flickr30k with 40% noise rate.

I thought you _followed the same training settings (e.g., optimizer, network architecture, all the hyperparameters, etc.) as NCR_ in this paper. Because NCR only adopts SGR as its backbone, and it involves 2 models for ensembling (2 co-learning SGR models). In contrast, at inference phase, this work will involve 4 models for ensembling (2 co-learning SGR models and 2 co-learning SAF models). Considering both NCR and DECL as previous works only adopt 2 models for ensembling, I am concerned that is it a fair comparison? Anyway, the SGR-only BiCro model does show a good performance over previous methods, which proves its effectiveness very well.

**Comment by ZihuaZhao** [comment]
> I ran multiple times with different seeds on Flickr30k with 40% noise rate and obtained very close results as follows:
> 
> ![image](https://user-images.githubusercontent.com/78494473/233528835-05625d0b-24b7-4e58-9f5b-36600af678d5.png)
> 
> I also tried to use hard labels for clean data just as the paper stated, and found that it will lead to much worse results (up to -6% decreases). Could the authors share the checkpoints of the 10-epoch warm-up models and the best co-learning models? Thanks a lot.

Hi, @xu5zhao and @ppj567 . I've been reproducing the results of BiCro under different noise rate as well. I've tested BiCro on both SGR and SGRAF. But still find a gap to the reported results. Especially under 60% noise, there's a gap of approximately 11 points in Recall sum score. @ppj567 , have you reproduced the similar results after applying SGRAF? And I'd like to ask the author @xu5zhao  for further model checkpoints under 20% noise and 60% noise. Thanks a lot.

**Comment by qxzha** [comment]
@ppj567 @ZihuaZhao @xu5zhao  I am concerned about this issue if fixed. I also retained BiCro using same settings on Flickr under 20%, 40% and 60% noise ratio, but I got a worse result than the reported ones. Should I need to modify some parameyers or codes ?
